web/docusaurus/src/components/CustomTOC/CustomTOC.tsx

```diff
@@ -86,10 +102,18 @@ export default function CustomTOC({ doc, onClick }) {
   }, [allDocsData, pathname]);
 
   const handleOnChange = (selectedOption) => {
+    // grab main docusarus configuration
+    const {siteConfig} = useDocusaurusContext();
     const { path } = selectedOption;
-    const { pluginId, version } = getPathKey(pathname);
-    const newPath = pathname.replace(`/${pluginId}/${version}`, path);
-    
+    const { localKey, pluginId, version } = getPathKey(pathname);
+    let newPath = undefined;
```

Comment:
not worth changing if you don't feel like it but I believe this defaults to undefined so you can just have 
```let newPath```